modulepath-tests/src/test/java/com/hazelcast/test/modulepath/SmokeModulePathTest.java

```diff
@@ -0,0 +1,110 @@
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Member;
+import com.hazelcast.instance.HazelcastInstanceFactory;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.hasItems;
+import org.junit.Test;
+
+/**
+ * Basic test which checks if correct Hazelcast modules are on the modulepath. It also checks that Hazelcast members and clients
+ * are able to start and form a cluster.
+ */
+public class SmokeModulePathTest {
+
+    /**
+     * Verify the Hazelcast JARs are on the modulepath.
+     */
+    @Test
+    public void testModulePath() {
+        String modulePath = System.getProperty("jdk.module.path");
+        assertNotNull("Module path was expected", modulePath);
+        assertTrue("Module path should contain hazelcast JAR",
+                modulePath.matches(".*hazelcast-[1-9][\\p{Alnum}\\-_\\.]+\\.jar.*"));
+        assertTrue("Module path should contain hazelcast-client JAR",
+                modulePath.matches(".*hazelcast-client-[\\p{Alnum}\\-_\\.]+\\.jar.*"));
+    }
+
+    /**
+     * Verify the Hazelcast modules with correct names are used.
+     */
+    @Test
+    public void testModuleNames() {
+        Set<String> hazelcastModuleNames = ModuleLayer.boot().modules().stream().map(Module::getName)
+                .filter(s -> s.contains("hazelcast")).collect(Collectors.toSet());
+        assertThat(hazelcastModuleNames, hasItems("com.hazelcast.core", "com.hazelcast.client", "com.hazelcast.tests"));
+    }
+
+    /**
+     * Verify Hazelcast members are able to start and form cluster. It also verifies the client is able to join and work with
+     * the cluster.
+     */
+    @Test
+    public void testCluster() {
+        Config config = new Config();
+
+        NetworkConfig networkConfig = config.getNetworkConfig();
+        networkConfig.getInterfaces().addInterface("127.0.0.1");
+        networkConfig.getJoin().getMulticastConfig().setEnabled(false);
+        networkConfig.getJoin().getTcpIpConfig().setEnabled(true).addMember("127.0.0.1");
+        try {
+            HazelcastInstance hz1 = Hazelcast.newHazelcastInstance(config);
+            HazelcastInstance hz2 = Hazelcast.newHazelcastInstance(config);
+            hz1.getMap("test").put("a", "b");
+            assertClusterSize(2, hz1, hz2);
+            HazelcastInstance client = HazelcastClient.newHazelcastClient();
+            assertEquals("b", client.getMap("test").get("a"));
+        } finally {
+            HazelcastInstanceFactory.terminateAll();
+            HazelcastClient.shutdownAll();
```

Comment:
I would shutdown the clients first, otherwise you might have disconnect exceptions in the log, which cause unwanted noise.

Reply:
Maybe this could also be moved to a `@After` teardown method, so the test code doesn't need to be in a `try finally` block.